### PR TITLE
Smart contracts: add .vscode settings.jsons

### PR DESCRIPTION
**Pull request by anchpop**
## Problem

Custom configuration was required to get vscode IDE features working with ocaml platform

## Solution

Add that configuration to .vscode/settings.json folders.

### CODE REVIEW

File: zinc/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "files.autoSave": "onFocusChange",
```
Comment (Zett98) [inline zinc/.vscode/settings.json:2]: maybe use afterDelay as in other two configs?

File: zinc/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        "root": "${workspaceFolder:zinc}"
+    },
+    "indentRainbow.ignoreErrorLanguages": [
+        "*"
+    ],
+    "files.insertFinalNewline": true,
+    "terminal.integrated.commandsToSkipShell": [
+        "-workbench.action.toggleSidebarVisibility"
+    ],
+    "terminal.integrated.defaultProfile.linux": "zsh"
```
Comment (Zett98) [inline zinc/.vscode/settings.json:18]: does everyone on the team use zsh?

File: deku/.vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "git.autofetch": true,
+    "ocaml.sandbox": {
+        "kind": "esy",
+        "root": "${workspaceFolder:deku}"
+    },
+    "indentRainbow.ignoreErrorLanguages": [
+        "*"
+    ],
+    "files.insertFinalNewline": true,
+    "terminal.integrated.commandsToSkipShell": [
```
Comment (Zett98) [inline deku/.vscode/settings.json:15]: isn't this should be specific to one's workflow?

File: zinc/.vscode/settings.json (hunk 2 above)
Comment (anchpop) [reply zinc/.vscode/settings.json:18]: Hmm, good point

File: deku/.vscode/settings.json (hunk 3 above)
Comment (anchpop) [reply deku/.vscode/settings.json:15]: Inserting the final newline is part of marigold style, is that what you were talking about?

File: zinc/.vscode/settings.json (hunk 1 above)
Comment (anchpop) [reply zinc/.vscode/settings.json:2]: Good point, thanks

File: deku/.vscode/settings.json (hunk 3 above)
Comment (Zett98) [reply deku/.vscode/settings.json:15]:
```
"terminal.integrated.commandsToSkipShell": [
```

**Comment by anchpop** [comment]
@Zett98 should be resolved

**Comment by Zett98** [comment]
LGTM

**Review by Zett98 — APPROVED** [review]

**Review by rosalogia — APPROVED** [review]

**Pull request merged**